### PR TITLE
Use a session to make nid persist after incorrect password.

### LINKED ISSUE #801: Pitch login bug
If you enter the wrong password, you are directed to user/login.

This means that once you enter the right password, you won't be signed up for the campaign, since you are no longer on the pitch page.

### PR DESCRIPTION
**Pull request by angaither**
Fixes #801

@desmondmorris @mshmsh5000 
IS THIS OK?
tumblr_m2e9dsu9tu1qdedm3o1_500

### CODE REVIEW

File: lib/modules/dosomething/dosomething_user/dosomething_user.module [hunk 1]
```diff
@@ -214,6 +216,7 @@ function _dosomething_user_add_campaign_data(&$form) {
     $obj = menu_get_object();
     // If we are viewing a campaign node:
     if (isset($obj->type) && $obj->type == 'campaign') {
+      $_SESSION['nid'] = $obj->nid;
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_user/dosomething_user.module:219]: Reading up on using session data in D7.  Seems we may need to manually start the session for anon users: 

"Drupal 7 sometimes does not automatically start sessions for anonymous user" http://drupal.stackexchange.com/questions/4851/how-do-i-save-session-data-for-the-anonymous-users

**Comment by DFurnes** [comment]
:heartbeat:

**Comment by aaronschachter** [comment]
This solution has a bug: As an anonymous user, I view a campaign pitch page with nid X.  If I then navigate to user/login on my own, to simply sign in, won't I then be automatically signed up for campaign X?

**Comment by mshmsh5000** [comment]
1. Session caching: worth looking at https://drupal.org/project/session_cache which has a smarter implementation.
2. Redirects: Would be much better, I think, to append `destination` param to user/login, so we can make this explicit.

**Pull request closed without merging**